engine/collect.go

```diff
@@ -79,7 +79,8 @@ func (c *collector) collect(ctx context.Context, server *autoscaler.Server) erro
 		Size:     server.Size,
 	}
 
-	client, err := c.client(server)
+	client, closer, err := c.client(server)
+	defer closer.Close()
```

Comment:
this is subject to a nil pointer exception, so you'll want the following:

```
if closer != nil {
  defer closer.Close()
}
```

(same feedback for other parts of the code that invokes the closer)